### PR TITLE
Fixed duplicate data fetching overview page + links

### PR DESCRIPTION
**Pull request by molebox**
This PR removes the duplicate overview link for data fetching, and fixes the corresponding links

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

**Comment by molebox** [comment]
We still get a 404 on the Overview page with these changes

File: docs/advanced-features/custom-error-page.md [hunk 1]
```diff
@@ -97,4 +97,4 @@ If you have a custom `Error` component be sure to import that one instead. `next
 
 ### Caveats
 
-- `Error` currently does not support Next.js [Data Fetching methods](/docs/basic-features/data-fetching.md) like [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) or [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering).
+- `Error` currently does not support Next.js [Data Fetching methods](/docs/basic-features/data-fetching.md) like [`getStaticProps`](/docs/basic-features/data-fetching/get-static-props.md) or [`getServerSideProps`](/docs/basic-features/data-fetching.md/get-server-side-props.md).
```
Comment (leerob) [inline docs/advanced-features/custom-error-page.md:100]:
```suggestion
- `Error` does not currently support Next.js [Data Fetching methods](/docs/basic-features/data-fetching.md) like [`getStaticProps`](/docs/basic-features/data-fetching/get-static-props.md) or [`getServerSideProps`](/docs/basic-features/data-fetching/get-server-side-props.md).
```

File: errors/gssp-export.md [hunk 2]
```diff
@@ -33,6 +33,6 @@ The `getServerSideProps` lifecycle is not compatible with `next export`, so you'
 
 ### Useful Links
 
-- [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization)
-- [`getStaticProps` documentation](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation)
-- [`exportPathMap` documentation](https://nextjs.org/docs/api-reference/next.config.js/exportPathMap)
+- [Automatic Static Optimization](/docs/advanced-features/automatic-static-optimization)
+- [`getStaticProps` documentation](/docs/basic-features/data-fetching/get-static-props.md)
+- [`exportPathMap` documentation](/docs/api-reference/next.config.js/exportPathMap)
```
Comment (leerob) [inline errors/gssp-export.md:38]:
```suggestion
- [Automatic Static Optimization](/docs/advanced-features/automatic-static-optimization.md)
- [`getStaticProps` documentation](/docs/basic-features/data-fetching/get-static-props.md)
- [`exportPathMap` documentation](/docs/api-reference/next.config.js/exportPathMap.md)
```

File: errors/page-data-collection-timeout.md [hunk 3]
```diff
@@ -15,4 +15,4 @@ When restarted it will retry all uncompleted jobs, but if a job was unsuccessful
 
 ### Useful Links
 
-- [`getStaticPaths`](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation)
+- [`getStaticPaths`](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths.md)
```
Comment (leerob) [inline errors/page-data-collection-timeout.md:18]:
```suggestion
- [`getStaticPaths`](/docs/basic-features/data-fetching/get-static-paths.md)
```

File: examples/with-mux-video/README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ This example uses:
 
 - [SWR](https://swr.vercel.app/) — dynamically changing the `refreshInterval` depending on if the client should be polling for updates or not
 - [`/pages/api`](pages/api) routes — a couple endpoints for making authenticated requests to the Mux API.
-- Dynamic routes using [`getStaticPaths` and `fallback: true`](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation), as well as dynamic API routes.
+- Dynamic routes using [`getStaticPaths` and `fallback: true`](/docs/basic-features/data-fetching/get-static-paths.md), as well as dynamic API routes.
```
Comment (leerob) [inline examples/with-mux-video/README.md:35]:
```suggestion
- Dynamic routes using [`getStaticPaths` and `fallback: true`](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths), as well as dynamic API routes.
```

File: examples/with-iron-session/README.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ It uses current best practices for authentication in the Next.js ecosystem and r
 
 **Features of the example:**
 
-- [API Routes](https://nextjs.org/docs/api-routes/dynamic-api-routes) and [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) examples.
+- [API Routes](/docs/api-routes/dynamic-api-routes) and [getServerSideProps](/docs/basic-features/data-fetching/get-server-side-props.md) examples.
```
Comment (leerob) [inline examples/with-iron-session/README.md:13]:
```suggestion
- [API Routes](https://nextjs.org/docs/api-routes/dynamic-api-routes) and [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) examples.
```

**Review by leerob — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Review by leerob — APPROVED** [review]

**Pull request merged**